### ISSUE TITLE
[visualization] No points displayed if pointcloud contains NaNs or Infinities

### ISSUE THREAD

**Issue by arjo129** [PointCloudLibrary/pcl#4233]
**Describe the bug**

If I pass a point cloud directly to the visuallizer which contains NaNs, no points are displayed. Not sure if this is a PCL problem or a VTK problem.

**Context**

Visuallize raw sensor data (which unfortunately does generate NaNs

**Expected behavior**

- Either a warning that tells me that I have NaNs and therefore the object cannot be rendered
- Silently ignore the NaNs and continue render of the finite points.

**Current Behavior**
- Blank screen, no warning, hours of debugging for solving a stupid problem.

**To Reproduce**
```C++
     pcl::PointCloud<pcl::PointXYZ> final_cloud1;
     final_cloud1.push_back(pcl::PointXYZ(0,0,0));
     final_cloud1.push_back(pcl::PointXYZ(INFINITY,0,0));
     pcl::PointCloud<pcl::PointXYZ>::Ptr final_points(new pcl::PointCloud<pcl::PointXYZ>);
     *final_points = final_cloud1;
     pcl::visualization::PointCloudColorHandlerCustom<pcl::PointXYZ> single_color1 (final_points, 0, 0, 200);
    
     viewer->removeAllPointClouds();
     viewer->addPointCloud(final_points, single_color1, "corners");
     viewer->setPointCloudRenderingProperties (pcl::visualization::PCL_VISUALIZER_POINT_SIZE, 2, "corners");
     viewer->setPointCloudRenderingProperties (pcl::visualization::PCL_VISUALIZER_POINT_SIZE, 2, "reference");
     viewer->resetCamera ();
```

**Your Environment (please complete the following information):**

 - OS: Ubuntu 18.04
 - Compiler: GCC 8.1
 - PCL Version 1.8

**Possible Solution**

Willing to raise a PR that would resolve this issue at a PCL level by warning if nans are in the pointcloud and removing them before hand. The other possibility is to add a warning when the Handlers check if the points are finite.

**Additional context**

**Comment by larshg** [comment]
Can you add a cloud with NaNs here for testing?

**Comment by JStech** [comment]
I was unable to reproduce this, using Ubuntu 18.04, PCL 1.8 (and 1.11), and g++ 7.5. The visualizer always started up just fine and showed the point.

**Comment by fahome10bd** [comment]
I am getting the same thing, if you get any solution, please give me some hope. I built both qt_colorize_cloud and qt_visualizer. However,  it is not showing any point where some cloud should be auto-generated and shown. 

Environment:
**OS: Windows 10 19043.1415
Compiler: MSVC++ version 14.29
PCL Version 1.11.1
Qt 5.14.2
vtk 8.2.0**

**Comment by larshg** [comment]
Sorry for the late answer.

I can verify that its not getting rendered when I use VTK 9.0.3, however I set `final_cloud1.is_dense` to **false**, then it renders it fine.
So it is important, that the code that generates the point cloud sets the is_dense flag accordingly, if it contains INIFINITY/NAN points.

@fahome10bd your issue seems to be different. If you still have an issue open a new ticket.

Closing as not a bug.
